src/no/runsafe/dergons/Dergon.java

```diff
@@ -116,145 +225,153 @@ public void e()
 		if (targetEntity != null && dergonLocation != null && random.nextFloat() < 0.2F)
 			((RunsafeFallingBlock) targetWorld.spawnFallingBlock(dergonLocation, Item.Unavailable.Fire)).setDropItem(false);
 
-		float f;
-		float f1;
-
-		if (world.isStatic)
+		if (world.isClientSide)
 		{
-			f = MathHelper.cos(by * 3.1415927F * 2.0F);
-			f1 = MathHelper.cos(bx * 3.1415927F * 2.0F);
-			if (f1 <= -0.3F && f >= -0.3F)
+			float animationPoint = (float) Math.cos(bv * PI * 2.0F);
+			float previousAnimationPoint = (float) Math.cos(bu * PI * 2.0F);
+			if (previousAnimationPoint <= -0.3F && animationPoint >= -0.3F)
 				world.a(locX, locY, locZ, "mob.enderdragon.wings", 5.0F, 0.8F + random.nextFloat() * 0.3F, false);
 		}
 
-		bx = by;
-		float f2;
+		bu = bv;
 
 		if (getHealth() <= 0.0F) // Check if the dragon is dead.
 		{
 			// If we're dead, play a random explosion effect at a random offset to it's corpse.
-			f = (random.nextFloat() - 0.5F) * 8.0F;
-			f1 = (random.nextFloat() - 0.5F) * 4.0F;
-			f2 = (random.nextFloat() - 0.5F) * 8.0F;
-			world.addParticle("largeexplode", locX + (double) f, locY + 2.0D + (double) f1, locZ + (double) f2, 0.0D, 0.0D, 0.0D);
+			world.addParticle(
+					EnumParticle.EXPLOSION_LARGE,
+					locX + (double) (random.nextFloat() - 0.5F) * 8.0F,
+					locY + (double) (random.nextFloat() - 0.5F) * 4.0F + 2.0D,
+					locZ + (double) (random.nextFloat() - 0.5F) * 8.0F,
+					0.0D,
+					0.0D,
+					0.0D
+			);
 		}
 		else
 		{
-			this.bN();
-			f = 0.2F / (MathHelper.sqrt(motX * motX + motZ * motZ) * 10.0F + 1.0F);
+			this.enderCrystalInteraction();
```

Comment:
I am 100% sure that that method deals with interacting with ender crystals from diving into the minecraft source code and spigot source code.  I'll remove it since dergons don't need to interact with ender crystals anyway.